lib/files.c

```diff
@@ -687,11 +687,34 @@ tool_rc files_save_ESYS_TR(ESYS_CONTEXT *ectx, ESYS_TR handle, const char *path)
         return rc == TPM2_RC_SUCCESS; \
     }
 
+#define LOAD_TYPE_FILE(type, name) \
+    bool files_load_##name##_file(FILE *f, const char *path, type *name) { \
+    \
+        UINT8 buffer[sizeof(*name)]; \
+        UINT16 size = sizeof(buffer); \
+        bool res = file_read_bytes_from_file(f, buffer, &size, path); \
+        if (!res) { \
+            return false; \
+        } \
+        \
+        size_t offset = 0; \
+        TSS2_RC rc = Tss2_MU_##type##_Unmarshal(buffer, size, &offset, name); \
+        if (rc != TSS2_RC_SUCCESS) { \
+            LOG_ERR("Error deserializing "str(name)" structure: 0x%x", rc); \
+            LOG_ERR("The input file needs to be a valid "xstr(type)" data structure"); \
+            return false; \
+        } \
+        \
+        return rc == TPM2_RC_SUCCESS; \
+    }
+
```

Comment:
Because LOAD_TYPE doesn't allow for the FILE * to be passed. LOAD_TYPE could now just call LOAD_TYPE_FILE and open/close the file around LOAD_TYPE_FILE, but that seemed just  over-refactoring and could be harder to follow.

Reply:
I could refactor the LOAD_TYPE macro to call functions defined by LOAD_TYPE_FILE, but then order matters, and we don't need to create _FILE entries for all of them now, making our library bigger for no reason.